### PR TITLE
Add support for splitting selections into lines

### PR DESCRIPTION
**Pull request by scholtzan**
This is a first attempt to solve https://github.com/google/xi-editor/issues/737

### CODE REVIEW

**Comment by scholtzan** [comment]
@cmyr: The bugs should be fixed now.

File: rust/core-lib/src/view.rs [hunk 1]
```diff
@@ -467,6 +468,38 @@ impl View {
         self.select_region(text, offset, SelRegion::new(start, end), multi_select);
     }
 
+    /// Splits current selections into lines.
+    fn do_split_selection_into_lines(&mut self, text: &Rope) {
+        let mut selection = Selection::new();
+
+        for region in self.selection.iter() {
+            if region.is_caret() {
+                selection.add_region(SelRegion::new(region.max(), region.max()));
```
Comment (cmyr) [inline rust/core-lib/src/view.rs]: maybe didn't notice this before, but for the record there's a `SelRegion::Caret` fn.

File: rust/core-lib/src/view.rs [hunk 2]
```diff
@@ -467,6 +468,38 @@ impl View {
...
+            } else {
+                let mut cursor = Cursor::new(&text, region.min());
+
+                while cursor.pos() < region.max() {
+                    let sel_start = cursor.pos();
+                    let end_of_line = match cursor.next::<LinesMetric>() {
```
Comment (cmyr) [inline rust/core-lib/src/view.rs]: This could possibly be another place for the pattern we discussed in your previous PR?

**Comment by eyelash** [comment]
Don't forget to document the new RPC in [frontend-protocol.md](https://github.com/google/xi-editor/blob/master/docs/docs/frontend-protocol.md).

**Comment by cmyr** [comment]
@scholtzan eyelash makes a good point... 😏

**Comment by scholtzan** [comment]
Done :)

**Pull request merged**